src/vivarium_public_health/utilities.py

```diff
@@ -123,3 +126,32 @@ def get_index_columns_from_lookup_configuration(
     for column in lookup_configuration["categorical_columns"]:
         index_columns.append(column)
     return index_columns
+
+
+def write_dataframe_to_csv(
+    observer: StratifiedObserver,
+    measure: str,
+    results: pd.DataFrame,
+    extra_cols: Optional[Dict[str, Any]] = {},
+):
+    results_dir = Path(observer.results_dir)
+    # Add extra cols
+    col_mapper = {"measure": measure}
+    col_mapper.update(extra_cols)
+    col_mapper.update(
+        {"random_seed": observer.random_seed, "input_draw": observer.input_draw}
+    )
+    for col, val in col_mapper.items():
+        if val is not None:
+            results[col] = val
```

Comment:
```suggestion
    col_mapper = {k: v for k, v in col_mapper.items() if v is not None}
```

Reply:
Decided not to change. It's the same number of lines and I think it's more readable (despite hating nested loops)